block/submitter.go

```diff
@@ -108,178 +137,112 @@ func (m *Manager) submitHeadersToDA(ctx context.Context) error {
 			m.logger.Debug("resetting DA layer submission options", "backoff", backoff, "gasPrice", gasPrice)
 		case coreda.StatusNotIncludedInBlock, coreda.StatusAlreadyInMempool:
 			m.logger.Error("DA layer submission failed", "error", res.Message, "attempt", attempt)
-			backoff = m.config.DA.BlockTime.Duration * time.Duration(m.config.DA.MempoolTTL) //nolint:gosec
+			backoff = m.config.DA.BlockTime.Duration * time.Duration(m.config.DA.MempoolTTL)
 			if m.gasMultiplier > 0 && gasPrice != -1 {
 				gasPrice = gasPrice * m.gasMultiplier
 			}
 			m.logger.Info("retrying DA layer submission with", "backoff", backoff, "gasPrice", gasPrice)
 		case coreda.StatusContextCanceled:
-			m.logger.Info("DA layer submission canceled", "attempt", attempt)
+			m.logger.Info("DA layer submission canceled due to context cancellation", "attempt", attempt)
 			return nil
+		case coreda.StatusTooBig:
+			fallthrough
 		default:
 			m.logger.Error("DA layer submission failed", "error", res.Message, "attempt", attempt)
 			backoff = m.exponentialBackoff(backoff)
 		}
-
-		attempt += 1
+		attempt++
 	}
 
-	if !submittedAllHeaders {
-		return fmt.Errorf(
-			"failed to submit all headers to DA layer, submitted %d headers (%d left) after %d attempts",
-			numSubmittedHeaders,
-			len(headersToSubmit),
-			attempt,
-		)
+	if !submittedAll {
```

Comment:
this seems like an issue? can you leave a comment on what happens here? does the system crash? does it try to resubmit?